tests/candev/Makefile

```diff
@@ -7,7 +7,36 @@ USEMODULE += isrpipe
 # define the CAN driver you want to use here
 CAN_DRIVER ?= native
 
-ifeq ($(CAN_DRIVER), PERIPH_CAN)
+ifeq ($(CAN_DRIVER), MCP2515)
+    FEATURES_REQUIRED = periph_gpio periph_spi periph_gpio_irq
+
+    USEMODULE += mcp2515
+
+	#set device paramters
+	TEST_MCP2515_SPI   ?= SPI_DEV\(0\)
+	TEST_MCP2515_SPI_CLK   ?= SPI_CLK_1MHZ
+	TEST_MCP2515_SPI_MODE   ?= SPI_MODE_0
+	TEST_MCP2515_CS    ?= GPIO_PIN\(PORT_B,0\)
+	TEST_MCP2515_RESET ?= GPIO_PIN\(PORT_B,2\)
+	TEST_MCP2515_INT ?= GPIO_PIN\(PORT_B,1\)
+	TEST_MCP2515_CLK ?=\(8000000ul\)
+	TEST_MCP2515_BITRATE ?= \(125000\)
+
+	#INT_PIN = PB3 --> requires AVR bank1 on avr. Change accordingly to used hardware.
+	USEMODULE += atmega_pcint0
+
+	# export parameters
+	CFLAGS += -DTEST_MCP2515_SPI=$(TEST_MCP2515_SPI)
```

Comment:
With the *mcp2515_params.h* approach, this should be as simple as
`CFLAGS += -DMCP2515_PARAM_SPI_CLK=$(TEST_MCP2515_SPI_CLK)`
And the device descriptor should be configured in there and not in [*main.c*](https://github.com/RIOT-OS/RIOT/pull/13624/files#diff-dea336f031c5b69baeaee003725acc05R41)

Reply:
Not entirely sure if I agree here (or I'm not entirely sure how the params approach is supposed to be used). 
The idea of having these `TEST_MCP2515_xxxx` parameters in the Makefile, is so that the user can easily adapt these settings to his own situation, without having to dig into the driver files. Because pretty much every user's situation will be unique, and these will always have to be adapted.

Reply:
Checkout [Default device configuration](http://doc.riot-os.org/driver-guide.html). I totally understand your motivation (personally I even tend think your approach makes is easier for developers), but the way that I propose seems to be the go-to solution in RIOT. Have a look at other driver tests under *tests/driver_*. Only few tests set these values in their Makefile which seem to be leftovers. My point is, I would like to be consistent here. 
Also, my personal impression is that developers with some experience immediately understand this concept (because they know that pin configurations vary from case to case). Less experienced developers usually need help with pin configurations anyway.

Reply:
I still think we cannot merge this way (please read [Default device configuration](http://doc.riot-os.org/driver-guide.html)). I could make a compromise if this was only valid for one specific board (and documented as such). But in that case I would skip the step introducing `TEST_MCP2515_SPI` and the like... Furthermore, `PORT_B` should be simply `1`